src/record_syscall.cc

```diff
@@ -2917,8 +2917,10 @@ static Switchable rec_prepare_syscall_arch(RecordTask* t,
       uint64_t result;
       vector<FileMonitor::Range> ranges;
       ranges.push_back(FileMonitor::Range(t->regs().arg2(), t->regs().arg3()));
-      if (t->fd_table()->emulate_read(
-              fd, t, ranges, t->get_io_offset(syscallno, t->regs()), &result)) {
+      int64_t offset = -1;
+      if (t->fd_table()->needs_offset(t, fd, false))
+        offset = t->get_io_offset(syscallno, t->regs());
```

Comment:
I should write a make target that runs clang-format and clang-tidy on my most recent commit ;).